gslab_scons/tests/test_release_function.py

```diff
@@ -40,37 +76,65 @@ def test_release_standard(self, mock_move, mock_make_archive, mock_copy,
         mock_getpass.return_value = 'test_token'
         mock_session.return_value = mock.MagicMock(post = mock.MagicMock(),
                                                    get  = mock.MagicMock())
-        mock_session.return_value.get.return_value.content = self.return_mock_release_data()
+
+        args = copy.deepcopy(standard_args)
+        release_data = helpers.mock_release_data(args)
+        mock_session.return_value.get.return_value.content = release_data
+
         # When org/repo refers to a nonexistent GitHub repository, 
         # make the release request fail. 
-        mock_session.return_value.post.side_effect = self.post_side_effect
+        mock_session.return_value.post.side_effect = fx.post_side_effect
+
+        # Give mock_upload a side effect preserving release assets listing
+        mock_upload.side_effect = fx.upload_asset_side_effect
+
+        args['zip_release'] = True
+
+        self.check_release(args, mock_session, mock_upload, mock_copy,
+                           mock_make_archive, mock_move)
+
+        # Test release() when DriveReleaseFiles is a string
+        args['DriveReleaseFiles'] = 'plot.pdf'
+        self.check_release(args, mock_session, mock_upload, mock_copy,
+                           mock_make_archive, mock_move)
 
-        # Add a side effect to mock_upload to preserve a listing of release assets
-        mock_upload.side_effect = self.upload_asset_side_effect
+        # Test release() when required directories don't exist
+        local_release = args['local_release']
+        mock_isdir.return_value = False
 
-        DriveReleaseFiles = ['paper.pdf', 'plot.pdf']
-        tools.release('test_version', 
-                      DriveReleaseFiles = DriveReleaseFiles, 
-                      local_release     = 'Google Drive/release/local_release/', 
-                      org               = 'org', 
-                      repo              = 'repo', 
-                      target_commitish  = 'test_branch', 
-                      zip_release       = True)
+        self.check_release(args, mock_session, mock_upload, mock_copy,
+                           mock_make_archive, mock_move)
+
+        # Did release() create the local_release subdirectory?
+        mock_makedirs.assert_any_call(local_release)
+        # Did it create the temporary directory for to-be-zipped release files?
+        mock_makedirs.assert_any_call('release_content')
+
```

Comment:
For `check_release()`? If so, no, as it is not a test method but a function that should receive mock objects as arguments. Let me give it a docstring...